src/Client/ConcordiumClient.cs

```diff
@@ -152,14 +185,326 @@ public ITransactionStatus GetBlockItemStatus(TransactionHash transactionHash)
     /// Get the status of and information about a specific block item given a transaction hash.
     /// </summary>
     /// <param name="transactionHash">Transaction Hash which is included in blocks returned.</param>
+    /// <param name="token">Cancellation token</param>
     /// <returns>Task which wraps a common return type for TransactionStatus.</returns>
-    public async Task<ITransactionStatus> GetBlockItemStatusAsync(TransactionHash transactionHash)
+    /// <exception cref="RpcException">RPC error occurred, access <see cref="RpcException.StatusCode"/> for more information.</exception>
+    public async Task<ITransactionStatus> GetBlockItemStatusAsync(TransactionHash transactionHash,
+        CancellationToken token = default)
     {
-        var blockItemStatus = await this.Raw.GetBlockItemStatusAsync(transactionHash.ToProto())
+        var blockItemStatus = await this.Raw.GetBlockItemStatusAsync(transactionHash.ToProto(), token)
             .ConfigureAwait(false);
 
         return TransactionStatusFactory.CreateTransactionStatus(blockItemStatus);
     }
 
+    /// <summary>
+    /// Get the information for the given account in the given block.
+    /// </summary>
+    /// <param name="accountIdentifier">Identifier of the account.</param>
+    /// <param name="blockHash">A hash of the block</param>
+    /// <param name="token">Cancellation token</param>
+    /// <returns>Account information at the given block.</returns>
+    /// <exception cref="RpcException">RPC error occurred, access <see cref="RpcException.StatusCode"/> for more information.</exception>
```

Comment:
We still need the documentation to include that we have put special meaning to the gRPC status code "Not found", which in this case means block or account was not found.
Similarly in the other endpoints taking `IBlockHashInput`.